app/components/extension-buttons.ts

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+export default function(app) {
+    app.component('extensionButtons', {
+      controller: [
+        '$scope', '$element', '$attrs', '$window',
+        function($scope, $element, $attrs, $window) {
+          this.isChromium = !!$window.chrome;
+          this.isFirefox = $window.navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
+
+          // from
+          // https://developer.mozilla.org/en/docs/Installing_Extensions_and_Themes_From_Web_Pages
+          $scope.firefoxAddonInstall = function(aEvent) {
+            for ( let a = aEvent.target; a.href === undefined; ) a = a.parentNode;
+            const params = {
+              'PaperHive': {
+                URL: aEvent.target.xpiUrl,
+                IconURL: aEvent.target.getAttribute('iconURL'),
+                toString: function () { return this.URL; }
+              }
+            };
+            $window.InstallTrigger.install(params);
+            return false;
+          };
```

Comment:
Since this doesn't work properly (I also wasted 20 minutes now): let's remove this code.